src/classes/DateLiteralConstant.cls

```diff
@@ -0,0 +1,225 @@
+public class DateLiteralConstant {
+    private String LAST_N = 'LAST_N_{0}: {1}';
```

Comment:
turns out you can. seems sort of counterintuitive but I guess they are final once constructed?